docs/core-concepts/state-management.mdx

```diff
@@ -4,12 +4,121 @@ title: State Management
 order: 100
 ---
 
-# State management
+There are many different ways to store and manage state in a Solid Start app. In this section we will discuss some of the most common ways to manage state in a project.
+
+## What is state?
+
+State is any information that is stored in your application. This can be anything from the current user's name to the current page the user is on. State can be stored in many different places, and can be accessed in many different ways.
+
+## Client-side state
+
+Client-side state is any state that is stored on the client's browser (and therefore, not available on the server). This can be stored in a variety of ways, including:
+
+- Local storage
+- Session storage
+- Cookies
+- Query parameters
+- Client-side stores
+
+### Local storage
+
+Local storage is a way to store data in the browser. This data is stored in the browser's local storage, and can be accessed by any page on the same domain. This data is persistent, and will remain even if the user closes their browser.
+
+When you need to access the `localStorage` API, you can either use a `createEffect` hook or a `createRouteAction` which gaurantees that the code will run in the browser.
+
+```tsx twoslash {19-21} filename="components/ThemePicker.tsx"
+// @include: lib
+// ---cut---
+import { createContext, createEffect, createSignal, useContext } from 'solid-start';
+import { Show, For } from 'solid-js';
+
+const ThemeContext = createContext(null);
+
+export function ThemeProvider() {
+  const INTITIAL_THEME = 'light';
+  const [theme, setTheme] = createSignal(INTIAL_THEME),
+    value = [
```

Comment:
Add `const` to this, there's a syntax error here

Reply:
uh oh, that means there is a [syntax error in the Solid docs too](https://www.solidjs.com/tutorial/stores_context?solved)! You are specifically talking about line 40, correct?

Reply:
Done